arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1913,42 +1956,122 @@ int ClusterInfo::createViewCoordinator(std::string const& databaseName,
-  AgencyWriteTransaction trans({delPlanCollection, incrementVersion},
-      {precondition, pre2});
-  res = ac.sendTransactionWithFailover(trans);
+  // FIXME do we really have to load plan even in case of error?
+  // Update our own cache
+  loadPlan();
+
+  int errorCode = TRI_ERROR_NO_ERROR;
 
   if (!res.successful()) {
-    AgencyCommResult ag = ac.getValues("");
+    if (res.errorCode() == int(arangodb::ResponseCode::PRECONDITION_FAILED)) {
+      errorMsg += "Precondition that view  with ID ";
+      errorMsg += viewID;
+      errorMsg += " already exist failed. Cannot create view.";
+
+      // Dump agency plan:
+      auto const ag = ac.getValues("/");
+
+      if (ag.successful()) {
+        LOG_TOPIC(ERR, Logger::CLUSTER)
+          << "Agency dump:\n"  << ag.slice().toJson();
+      } else {
+        LOG_TOPIC(ERR, Logger::CLUSTER)
+          << "Could not get agency dump!";
+      }
+    } else {
+      errorMsg += std::string("file: ") + __FILE__ +
+                  " line: " + std::to_string(__LINE__);
+      errorMsg += " HTTP code: " + std::to_string(res.httpCode());
+      errorMsg += " error message: " + res.errorMessage();
+      errorMsg += " error details: " + res.errorDetails();
+      errorMsg += " body: " + res.body();
+    }
+
+    errorCode = TRI_ERROR_CLUSTER_COULD_NOT_REMOVE_COLLECTION_IN_PLAN; // FIXME COULD_NOT_REMOVE_VIEW_IN_PLAN
+  }
+
+  events::DropView(viewID, errorCode);
+  return errorCode;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief set view properties in coordinator
+////////////////////////////////////////////////////////////////////////////////
+
+Result ClusterInfo::setViewPropertiesCoordinator(
+    std::string const& databaseName,
+    std::string const& viewID,
+    VPackSlice const& json
+) {
+  AgencyComm ac;
+
+  auto res = ac.getValues("Plan/Views/" + databaseName + "/" + viewID);
+
+  if (!res.successful()) {
+    return { TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND };
+  }
+
+  auto const view = res.slice()[0].get(
+    { AgencyCommManager::path(), "Plan", "Views", databaseName, viewID }
+  );
+
+  if (!view.isObject()) {
+    auto const ag = ac.getValues("");
+
     if (ag.successful()) {
       LOG_TOPIC(ERR, Logger::CLUSTER) << "Agency dump:\n"
                                       << ag.slice().toJson();
     } else {
       LOG_TOPIC(ERR, Logger::CLUSTER) << "Could not get agency dump!";
     }
+
+    return { TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND };
   }
 
-  // Update our own cache:
-  loadPlan();
+  res.clear();
 
-  events::DropView(viewID, TRI_ERROR_NO_ERROR);
-  return TRI_ERROR_NO_ERROR;
+  AgencyWriteTransaction const trans{
+    // operations
+    {
+      { "Plan/Views/" + databaseName + "/" + viewID, AgencyValueOperationType::SET, json },
+      { "Plan/Version", AgencySimpleOperationType::INCREMENT_OP }
+    },
+    // preconditions
+    { "Plan/Databases/" + databaseName, AgencyPrecondition::Type::EMPTY, false }
```

Comment:
Should we not have a precondition here that the value is not yet changed? Therefore the loadPlan is probably sensible to get the latest state.

Reply:
Currently this is similar to how DDL for collections is working. In case if you think that is OK to `loadPlan` even in case of error (e.g. to refresh something) we can easily get rid of this precondition. On the other hand it worth nothing to preserve it there. I would argue for having similar approach for collections and views.